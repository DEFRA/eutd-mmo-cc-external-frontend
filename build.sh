#/bin/bash
curl -O https://nodejs.org/dist/v12.18.4/node-v12.18.4-linux-x64.tar.xz
tar -xf node-v12.18.4-linux-x64.tar.xz --directory /tmp
export PATH="/tmp/node-v12.18.4-linux-x64/bin:$PATH"
rm node-v12.18.4-linux-x64.tar.xz
npm ci
npm run test-vstack
rm -rf /tmp/node-v12.18.4-linux-x64