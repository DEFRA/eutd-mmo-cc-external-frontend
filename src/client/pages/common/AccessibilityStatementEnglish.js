import React from 'react';
import {
  GridCol,
  GridRow,
  Header,
  ListItem,
  Main,
  UnorderedList,
  Table
} from 'govuk-react';
import PageTitle from '../../components/PageTitle';
import LinkTargetBlank from '../../components/elements/LinkTargetBlank';

export const AccessibilityStatementEnglish = () => (
  <Main>
    <PageTitle title='Accessibility - GOV.UK' />
    <GridRow>
      <GridCol>
        <Header level="1">Accessibility statement for the Fish Exports service</Header>
        <p>This service is part of the wider <LinkTargetBlank href="https://www.gov.uk" ariaLabel="Opens link for GOV.UK" text="GOV.UK"/> website. There’s a separate &nbsp;
          <LinkTargetBlank href="https://www.gov.uk/help/accessibility-statement" ariaLabel="Opens link for accessibility statement for main gov.uk site" text="accessibility statement for the main GOV.UK website."/>
        </p>
        <p>This page only contains information about the Fish Exports service, available at: <br />
          <LinkTargetBlank href="/" ariaLabel="Opens link for information on fish export service" text="https://manage-fish-exports.service.gov.uk/" />
        </p>
        <br />
        <Header level="2" size="MEDIUM">Using this service</Header>
        <p>This service is run by the Marine Management Organisation. We want as many people as possible to be able to use this service. For example, that means you should be able to:</p>
        <UnorderedList className="ul" listStyleType="disc">
          <ListItem>change colours, contrast levels and fonts</ListItem>
          <ListItem>zoom in up to 300% without the text spilling off the screen</ListItem>
          <ListItem>get from the start of the service to the end using just a keyboard</ListItem>
          <ListItem>get from the start of the service to the end using speech recognition software</ListItem>
          <ListItem>listen to the service using a screen reader (including the most recent versions of JAWS, NVDA and VoiceOver)</ListItem>
        </UnorderedList>
        <br/>
        <p>We’ve also made the text in the service as simple as possible to understand.</p>
        <p><LinkTargetBlank href="https://mcmw.abilitynet.org.uk/" ariaLabel="Opens link about using your device if you have disability" text="AbilityNet" /> has advice on making your device easier to use if you have a disability.</p>
        <Header level="2" size="MEDIUM">How accessible this service is</Header>
        <p>We know some parts of this website are not fully accessible.</p>
        <UnorderedList className="ul" listStyleType="disc">
          <ListItem>There are compatibility issues with some assistive technology</ListItem>
          <ListItem>Some navigation is inconsistent or not clearly connected to its content</ListItem>
          <ListItem>There are text resizing issues on certain pages</ListItem>
          <ListItem>Accessible names are missing from some buttons</ListItem>
          <ListItem>Some error messages require clarification</ListItem>
          <ListItem>There is incorrect heading structure on certain pages</ListItem>
          <ListItem>There is an incorrect page title implementation on the Catch Certificates Add landings page</ListItem>
          <ListItem>Labels can become removed from their content on the Catch Certificates Add landings page</ListItem>
        </UnorderedList>
        <br/>
        <Header level="2" size="MEDIUM">Feedback and contact information</Header>
        <p>If you have difficulty using this service, contact us by:</p>
        <UnorderedList className="ul" listStyleType="disc">
          <ListItem>call <a href="tel:0330 159 1989 ">0330 159 1989</a></ListItem>
        </UnorderedList>
        <br />
        <p>As part of providing this service, we may need to send you messages or documents. We’ll ask you how you want us to send messages or documents to you but contact us if you need them in a different format. For example, large print, audio recording or braille.</p>
        <Header level="2" size="MEDIUM"> Reporting accessibility problems with this service</Header>
        <p>We’re always looking to improve the accessibility of this service. If you find any problems that are not listed on this page or think we’re not meeting accessibility requirements, contact: Dom Horsfall (<a href="mailto:dominic.Horsfall@marinemanagement.org.uk">dominic.Horsfall@marinemanagement.org.uk</a>)</p>
        <br />
        <Header level="2" size="MEDIUM">Enforcement procedure</Header>
        <p>The Equality and Human Rights Commission (EHRC) is responsible for enforcing the Public Sector Bodies (Websites and Mobile Applications) (No. 2) Accessibility Regulations 2018 (the ‘accessibility regulations’). If you’re not happy with how we respond to your complaint, &nbsp;
          <LinkTargetBlank href="https://www.equalityadvisoryservice.com/" ariaLabel="Opens link to information about equality advisory service" text="contact the Equality Advisory and Support Service (EASS)." />
        </p>
        <br />

        <Header level="2" size="MEDIUM">Contacting us by phone or visiting us in person</Header>
        <p>We provide a text relay service for people who are deaf, hearing impaired or have a speech impediment.</p>
        <p>Our offices have audio induction loops, or if you contact us before your visit, we can arrange a British Sign Language (BSL) interpreter to help you complete the service in person.</p>
        <p>Find out how to contact us: &nbsp;
         <LinkTargetBlank href="https://www.gov.uk/contact-local-marine-management-organisation" ariaLabel="Opens link to Contact your local Marine Management Organisation office" text="Contact your local Marine Management Organisation office" />
        </p>


        <br />
        <Header level="2" size="MEDIUM">Technical information about this service’s accessibility</Header>
        <p>The Marine Management Organisation is committed to making this service accessible, in accordance with the Public Sector Bodies (Websites and Mobile Applications) (No. 2) Accessibility Regulations 2018.</p>

        <br />
        <Header level="3" size="MEDIUM">Compliance status</Header>
        <p>{'This website is partially compliant with the '}
        <LinkTargetBlank href="https://www.w3.org/TR/WCAG21/" ariaLabel="Opens link to Web Content Accessibility Guidelines" text="Web Content Accessibility Guidelines version 2.1 AA standard" />, due to the non-compliances listed below.</p>
        <br />

        <Header level="2" size="MEDIUM">Non accessible content</Header>
        <p>The content listed below is non-accessible for the following reasons.</p>

        <br />
        <Header level="3" size="MEDIUM">Non-compliance with the accessibility regulations</Header>
        <p>The service has identified the following issues:</p>
        <Table
         head={(
          <Table.Row>
            <Table.CellHeader>Reference</Table.CellHeader>
            <Table.CellHeader>Issue</Table.CellHeader>
            <Table.CellHeader>Location</Table.CellHeader>
            <Table.CellHeader>WCAG ref.</Table.CellHeader>
          </Table.Row>
         )}
         body={
          <>
            <Table.Row>
             <Table.Cell><strong>AA1/A-05a</strong></Table.Cell>
             <Table.Cell><strong>{'Month-by-month navigation isn\'t visually or semantically connected to the \'Completed\' data above it.'}</strong></Table.Cell>
             <Table.Cell>CC,PS,SD - Exporter dashboard</Table.Cell>
             <Table.Cell>1.3.2</Table.Cell>
           </Table.Row>
           <Table.Row>
             <Table.Cell><strong>AA1/A-05b</strong></Table.Cell>
             <Table.Cell><strong>{'The \'<<\' and \'>>\' links to future/previous months are not explained or semantically presented in way that is accessible.'}</strong></Table.Cell>
             <Table.Cell>CC,PS,SD - Exporter dashboard</Table.Cell>
             <Table.Cell>1.3.2</Table.Cell>
           </Table.Row>
           <Table.Row>
             <Table.Cell><strong>AA1/A-23</strong></Table.Cell>
             <Table.Cell><strong>Incorrect heading structure</strong></Table.Cell>
             <Table.Cell>CC,PS,SD - Exporter dashboard</Table.Cell>
             <Table.Cell>1.3.1</Table.Cell>
           </Table.Row>
           <Table.Row>
             <Table.Cell><strong>AA1/A-07</strong></Table.Cell>
             <Table.Cell><strong>{'The text can be resized, but the way the form is handled means that the field widths don\'t adequately accommodate the resized text, partially obscuring the content.'}</strong></Table.Cell>
             <Table.Cell>CC - Add landings</Table.Cell>
             <Table.Cell>1.4.4</Table.Cell>
           </Table.Row>
           <Table.Row>
             <Table.Cell><strong>AA1/A-08</strong></Table.Cell>
             <Table.Cell><strong>On text resizing, the form does not reflow on resizing, causing some content to be part hidden. Similarly, on smaller screen sizes, parts of the form appear off screen.</strong></Table.Cell>
             <Table.Cell>CC - Add landings</Table.Cell>
             <Table.Cell>1.4.10</Table.Cell>
           </Table.Row>
           <Table.Row>
             <Table.Cell><strong>AA1/A-09</strong></Table.Cell>
             <Table.Cell><strong>Date picker issues.</strong></Table.Cell>
             <Table.Cell>CC,PS,SD</Table.Cell>
             <Table.Cell>2.1.1</Table.Cell>
           </Table.Row>
           <Table.Row>
             <Table.Cell><strong>AA1/A-10</strong></Table.Cell>
             <Table.Cell><strong>Typeahead issues - ‘Add landings for each product’.</strong></Table.Cell>
             <Table.Cell>CC,PS,SD</Table.Cell>
             <Table.Cell>2.1.1</Table.Cell>
           </Table.Row>
           <Table.Row>
             <Table.Cell><strong>AA1/A-29</strong></Table.Cell>
             <Table.Cell><strong>Page title issue with add landings - incorrect title element</strong></Table.Cell>
             <Table.Cell>CC - Add landings</Table.Cell>
             <Table.Cell>2.4.2</Table.Cell>
           </Table.Row>
           <Table.Row>
             <Table.Cell><strong>AA1/A-17</strong></Table.Cell>
             <Table.Cell><strong>Accessible names are missing from the 'Edit' and 'Remove' buttons.</strong></Table.Cell>
             <Table.Cell>CC - Add landings</Table.Cell>
             <Table.Cell>2.5.3</Table.Cell>
           </Table.Row>
           <Table.Row>
             <Table.Cell><strong>AA1/A-30</strong></Table.Cell>
             <Table.Cell><strong>Navigation issues - inconsistent main navigation</strong></Table.Cell>
             <Table.Cell>CC,PS,SD - Exporter dashboard</Table.Cell>
             <Table.Cell>3.2.3</Table.Cell>
           </Table.Row>
           <Table.Row>
             <Table.Cell><strong>AA1/A-18</strong></Table.Cell>
             <Table.Cell><strong>Instances of generic error messaging.</strong></Table.Cell>
             <Table.Cell>CC - Add landings</Table.Cell>
             <Table.Cell>3.3.1</Table.Cell>
           </Table.Row>
           <Table.Row>
             <Table.Cell><strong>AA1/A-33</strong></Table.Cell>
             <Table.Cell><strong>Landings form labels disappear from view when multiple landings are added.</strong></Table.Cell>
             <Table.Cell>CC - Add landings</Table.Cell>
             <Table.Cell>3.3.2</Table.Cell>
           </Table.Row>
           <Table.Row>
             <Table.Cell><strong>AA1/A-34</strong></Table.Cell>
             <Table.Cell><strong>Validation is specific regarding “transportation details” but does not inform the user what they need to do to correct their mistakes.</strong></Table.Cell>
             <Table.Cell>CC,SD - Add transportation details</Table.Cell>
             <Table.Cell>3.3.1</Table.Cell>
           </Table.Row>
           <Table.Row>
             <Table.Cell><strong>AA1/A-01</strong></Table.Cell>
             <Table.Cell><strong>Drop-down pick lists not accessible with Dragon Naturallyspeaking.</strong></Table.Cell>
             <Table.Cell>CC - Add products</Table.Cell>
             <Table.Cell>4.1.1</Table.Cell>
           </Table.Row>
           <Table.Row>
             <Table.Cell><strong>AA1/A-02</strong></Table.Cell>
             <Table.Cell><strong>Date cannot be chosen using Dragon Naturallyspeaking.</strong></Table.Cell>
             <Table.Cell>CC - Add landings</Table.Cell>
             <Table.Cell>4.1.1</Table.Cell>
           </Table.Row>
           <Table.Row>
             <Table.Cell><strong>AA1/A-03</strong></Table.Cell>
             <Table.Cell><strong>Drop-down pick lists not accessible with Dragon Naturallyspeaking.</strong></Table.Cell>
             <Table.Cell>CC - Add landings</Table.Cell>
             <Table.Cell>4.1.1</Table.Cell>
           </Table.Row>
           <Table.Row>
             <Table.Cell><strong>AA1/A-20</strong></Table.Cell>
             <Table.Cell><strong>Compatibility issues with Dragon Naturally Speaking.</strong></Table.Cell>
             <Table.Cell>CC,PS,SD</Table.Cell>
             <Table.Cell>4.1.1</Table.Cell>
           </Table.Row>
           <Table.Row>
             <Table.Cell><strong>AA1/A-21</strong></Table.Cell>
             <Table.Cell><strong>Navigation issues - missing ARIA roles</strong></Table.Cell>
             <Table.Cell>CC,PS,SD</Table.Cell>
             <Table.Cell>4.1.2</Table.Cell>
           </Table.Row>
           <Table.Row>
             <Table.Cell><strong>AA1/A-35</strong></Table.Cell>
             <Table.Cell><strong>There is more than one type of format used for date pickers. This version is not keyboard-accessible.</strong></Table.Cell>
             <Table.Cell>SD - Add product</Table.Cell>
             <Table.Cell>4.1.1</Table.Cell>
           </Table.Row>
          </>
         }
        />
        <br />
        <p>CC = Catch Certificates, PS = Processing Statements, SD = Storage Documents</p>
        <p>We intend to fix these problems by as soon as possible.</p>

        <br />
        <Header level="3" size="MEDIUM">Disproportionate burden</Header>
        <p>For catch certificates:</p>
        <UnorderedList className="ul" listStyleType="disc">
          <ListItem>the forms for direct landings are only available in PDF format</ListItem>
        </UnorderedList>
        <br/>
        <p>These PDF forms have passed WCAG AA standards, however, we do intend to replace them with HTML forms in the long term.</p>

        <Header level="2" size="MEDIUM">What we’re doing to improve accessibility</Header>
        <p>As part of continuous improvement, the service aims to:</p>
        <UnorderedList className="ul" listStyleType="disc">
          <ListItem>address and fix identified, WCAG 2.1 AA non-compliant accessibility issues (priority 1)</ListItem>
          <ListItem>carry out continuous research to make sure you can complete the service first time, and make improvements where necessary (priority 2)</ListItem>
          <ListItem>continue to research and test the service to ensure your needs are addressed (priority 3)</ListItem>
        </UnorderedList>
        <br />
        <Header level="2" size="MEDIUM">Preparation of this accessibility statement</Header>
        <p>This statement was prepared on 3 June 2021. It was last reviewed on 3 June 2021.</p>
        <p>This service was last tested on 21 September 2020. The test was carried out by Defra’s Digital, Data and Technology Services accessibility team.</p>
        <br />
      </GridCol>
    </GridRow>
  </Main>
);