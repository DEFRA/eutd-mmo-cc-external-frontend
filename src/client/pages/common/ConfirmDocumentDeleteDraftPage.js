import React from 'react';
import {withRouter} from 'react-router';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';
import _ from 'lodash';

import {
  saveConfirmDocumentDelete,
  addConfirmDocumentDelete,
  dispatchApiCallFailed
} from '../../actions';
import {
  Main,
  BackLink,
  Header,
  GridRow,
  GridCol,
  MultiChoice
} from 'govuk-react';

import PageTitle from '../../components/PageTitle';
import Form from '../../components/elements/Form';
import HelpLink from '../../components/HelpLink';
import ContinueButton from '../../components/elements/ContinueButton';
import ErrorIsland from '../../components/elements/ErrorIsland';

import { scrollToErrorIsland, onHandleErrorClick} from '../utils';
import SelectRadio from '../../components/elements/SelectRadio';
import PageTemplateWrapper from '../../components/PageTemplateWrapper';
import { withTranslation } from 'react-i18next';
import i18n from '../../../../src/i18n';
class ConfirmDocumentDeleteDraftPage extends React.Component {

  componentDidMount(){
    i18n.changeLanguage();
    this.forceUpdate();
  }

  onChange = e => {
    const confirmDocumentDelete = _.cloneDeep(
      this.props.confirmDocumentDelete || {}
    );
    confirmDocumentDelete.documentDelete = e.target.value;
    this.props.addConfirmDocumentDelete(confirmDocumentDelete);
  };

  onSubmit = async e => {
    e.preventDefault();

    try {
      const {path, journey, previousUri, nextUri} = this.props.route;
      const documentNumber = this.props.match.params['documentNumber'];
      const documentDelete = await this.props.saveConfirmDocumentDelete(path, nextUri, journey, documentNumber);
      const to = (documentDelete && documentDelete === 'Yes') ? nextUri : previousUri;

      this.props.history.push(to);
    } catch (error) {
      scrollToErrorIsland();
    }
  };

  render() {
    const {documentDeleteError, errors} = this.props.errors;
    const {confirmDocumentDelete = {}, route, match, t} = this.props;
    const {path, previousUri, nextUri, journey, journeyText} = route;
    const title = t(`${journey}DeleteCatchCertificateDeleteDraftJourneytext`, {journeyText: t(journeyText)})
    const h1Text =  t(`${journey}DeleteCatchCertificateDraftDeleteConfirmationText`, {journeyText: t(journeyText)})
    const currentUri = path.replace(':documentNumber', match.params.documentNumber);
    return (
      <Main>
        {errors && (
          <ErrorIsland
            errors={errors.map(err => ({
              message: t(err.text),
              key: err.targetName
            }))}
            onHandleErrorClick={onHandleErrorClick}
          />
        )}

        <PageTitle
          title={`${!_.isEmpty(errors) ? 'Error: ' : ''}${title} - ${t(this.props.route.title)}`}
        />
        <GridRow>
            <BackLink onClick={e => { e.preventDefault(); this.props.history.push(previousUri); }} href={previousUri}>
            {t('commonBackLinkBackButtonLabel')}
              </BackLink>
        </GridRow>

        <Form action='/orchestration/api/v1/confirm-document-delete' currentUrl={currentUri} onSubmit={this.onSubmit}>
          <GridRow>
            <GridCol columnTwoThirds>
              <GridRow>
                <Header level='1'>
                  <legend>{h1Text}</legend>
                </Header>
              </GridRow>
              <MultiChoice meta={{touched: t(documentDeleteError), error: t(documentDeleteError)}} id='documentDelete'>
                <fieldset>
                  <SelectRadio id="documentDeleteYes" checked={confirmDocumentDelete.documentDelete === 'Yes'} value='Yes' name='documentDelete' inline onChange={this.onChange}>{t('Yes')}</SelectRadio>
                  <SelectRadio id="documentDeleteNo" checked={confirmDocumentDelete.documentDelete === 'No'} value='No' name='documentDelete' inline onChange={this.onChange}>{t('No')}</SelectRadio>
                </fieldset>
              </MultiChoice>
            </GridCol>
          </GridRow>
          <GridRow>
            <GridCol columnTwoThirds>
              <ContinueButton type='submit' id='continue'>{t('commonContinueButtonSaveAndContinueButton')}</ContinueButton>
            </GridCol>
          </GridRow>
          <input type="hidden" name="journey" value={journey} />
          <input type="hidden" name="currentUri" value={currentUri} />
          <input type="hidden" name="nextUri" value={nextUri} />
          <input type="hidden" name="previousUri" value={previousUri} />
        </Form>
        <HelpLink journey={route.journey} />
      </Main>
    );
  }
}

function mapStateToProps(state) {
  return {
    errors: state.errors,
    confirmDocumentDelete: state.confirmDocumentDelete
  };
}

function loadData(store) {
  if (Object.keys(this.queryParams).length !== 0) {
    store.dispatch(dispatchApiCallFailed(JSON.parse(this.queryParams.error)));
  }
}

export const component = withRouter(
  connect(
    mapStateToProps,
    {
      addConfirmDocumentDelete,
      saveConfirmDocumentDelete
    }
  )(withTranslation()(ConfirmDocumentDeleteDraftPage))
);

ConfirmDocumentDeleteDraftPage.propTypes = {
  confirmDocumentDelete: PropTypes.object.isRequired,
  errors: PropTypes.object
};

export default {
  loadData,
  component: PageTemplateWrapper(component)
};
