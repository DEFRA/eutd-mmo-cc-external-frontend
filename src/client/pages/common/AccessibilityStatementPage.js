import React from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { getAllUserAttributes } from '../../actions/userAttributes.actions';
import { AccessibilityStatementEnglish } from './AccessibilityStatementEnglish';
import { AccessibilityStatementWelsh } from './AccessibilityStatementWelsh';

const AccessibilityStatementPage = (props) => {
  let userSelectedLanguage;
  const userAttributeLanguage = props.userAttributes.find((val) => val.name === 'language');
  
  if (userAttributeLanguage) {
    userSelectedLanguage = userAttributeLanguage.value;
  }

  return (
    userSelectedLanguage === 'cy_UK' 
      ? <AccessibilityStatementWelsh />
      : <AccessibilityStatementEnglish />
  );
};

const mapStateToProps = state => {
  return {
    userAttributes: state.userAttributes
  };
};

const loadData = store => {
  return store.dispatch(getAllUserAttributes());
};

export default {
  loadData,
  component: withRouter(connect(mapStateToProps,
    {
      getAllUserAttributes
    })(AccessibilityStatementPage))
};
