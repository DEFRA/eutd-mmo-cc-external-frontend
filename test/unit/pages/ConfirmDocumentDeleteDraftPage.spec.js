import { mount } from 'enzyme/build';
import { Provider } from 'react-redux';
import * as React from 'react';
import _ from 'lodash';
import * as ConfirmDocumentDeleteDraftPage from '../../../src/client/pages/common/ConfirmDocumentDeleteDraftPage';
import ConfirmDocumentDeleteDraftPageWrapper from '../../../src/client/pages/common/ConfirmDocumentDeleteDraftPage';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { MemoryRouter } from 'react-router-dom';
import {
  saveConfirmDocumentDelete,
  addConfirmDocumentDelete,
  dispatchApiCallFailed
} from '../../../src/client/actions';
import { act } from 'react-dom/test-utils';
import { render } from '@testing-library/react';

jest.mock('../../../src/client/actions');

const mockStore = configureStore([thunk.withExtraArgument({
  orchestrationApi: {
    post: () => {
      return new Promise((res) => {
        res({});
      });
    },
    get: () => {
      return new Promise((res) => {
        res({});
      });
    }
  }
})]);

describe('Confirm Document Delete Draft page', () => {
  let wrapper;


  const storeA = {
    addConfirmDocumentDelete  : {},
    saveConfirmDocumentDelete : {},
    errors: { errors: [] },
    t: jest.fn()
  };

  beforeEach(() => {
    saveConfirmDocumentDelete.mockReturnValue({ type: 'SAVE_CONFIRM_DELETE' });
    addConfirmDocumentDelete.mockReturnValue({ type: 'ADD_CONFIRM_DELETE' });
    dispatchApiCallFailed.mockReturnValue({ type: 'CALL_FAILED' });
    const store = mockStore({
        addConfirmDocumentDelete  : {},
        saveConfirmDocumentDelete : {},
        errors                    : {},
        t                         : jest.fn()
    });

    const props = {
        errors : {},
        route: {
          title                 : 'Create a UK catch certificate - GOV.UK',
          nextUri               : '/create-catch-certificate/what-are-you-exporting',
          journey               : 'catchCertificate',
          previousUri           : '/create-catch-certificate/catch-certificates',
          path                  : '/create-catch-certificate/catch-certificates',
          history               : [],
          journeyText           : 'catch certificate'
        },
        t: jest.fn()
    };


    wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
            <ConfirmDocumentDeleteDraftPage.default.component {...props}/>
        </MemoryRouter>
      </Provider>
    );
  });

  it('should load successfully', () => {
    expect(wrapper).toBeDefined();
  });

  it('should handle submit event', () => {
    wrapper.find('form').simulate(
      'submit',
      {preventDefault() {}}
    );
  });

  it('should handle on change events', () => {
    wrapper.find('input[name="documentDelete"][id="documentDeleteYes"]').simulate('change', {target: {name: 'documentDelete',  value: 'Yes'}});
  });

  it('should find the legend tags', () => {
    expect(wrapper.find('legend')).toHaveLength(1);
  });

  it('should call the onClick of the back link', () => {
    const mockPreventDefault = jest.fn();

    const baclLinkEl = wrapper.find('BackLink');
    baclLinkEl.simulate('click', {
      preventDefault: mockPreventDefault
    });

    expect(mockPreventDefault).toHaveBeenCalled();
  });

    it('should handle submit event and push history', () => {
     act(() => wrapper.find('form').props().onSubmit(
      {preventDefault() {}}
    ));
  });
  
  it('should throw an error if the user has not made a selection', () => {
    const store = _.cloneDeep(storeA);
    const props = {
      errors : {},
      route: {
        title                 : 'Create a UK catch certificate - GOV.UK',
        nextUri               : '/create-catch-certificate/what-are-you-exporting',
        journey               : 'catchCertificate',
        previousUri           : '/create-catch-certificate/catch-certificates',
        path                  : '/create-catch-certificate/catch-certificates',
        history               : [],
        journeyText           : 'catch certificate'
      },
      t: jest.fn()
  };
    store.errors = {
      documentDelete: 'Select yes if you want to delete the current document and start a new one',
      errors: [{ 'targetName': 'documentDelete', 'text': 'Select yes if you want to delete the current document and start a new one' }]
    };

    wrapper = mount(
      <Provider store={mockStore(store)}>
        <MemoryRouter>
        <ConfirmDocumentDeleteDraftPage.default.component {...props}/>
        </MemoryRouter>
      </Provider>
    );

    let ErrorIsland = wrapper.find('div#errorIsland');
    let errorSummary = wrapper.find('ul.error-summary-list li a');
    expect(ErrorIsland.exists()).toBeTruthy();
    expect(wrapper.find('ErrorIsland').prop('errors')[0].message).toBe('Select yes if you want to delete the current document and start a new one');
});

  it('should take a snapshot of the whole page', () => {
    const { container } = render(wrapper);
    expect(container).toMatchSnapshot();
  });
});

describe('loadData', () => {
  const store = {
    dispatch: jest.fn()
  };

  const documentNumber = 'some-document-number';
  const queryParams = {
    error: '{"x":5,"y":6}'
  };

  it('will call all methods needed to load the component', () => {
    ConfirmDocumentDeleteDraftPageWrapper.documentNumber = documentNumber;
    ConfirmDocumentDeleteDraftPageWrapper.queryParams = {};

    ConfirmDocumentDeleteDraftPageWrapper.loadData(store);

    expect(store.dispatch).not.toHaveBeenCalled();
  });

  it('will call all methods needed to load the component when given query params', () => {
    ConfirmDocumentDeleteDraftPageWrapper.documentNumber = documentNumber;
    ConfirmDocumentDeleteDraftPageWrapper.queryParams = queryParams;

    ConfirmDocumentDeleteDraftPageWrapper.loadData(store);

    expect(store.dispatch).toHaveBeenCalled();
  });
});