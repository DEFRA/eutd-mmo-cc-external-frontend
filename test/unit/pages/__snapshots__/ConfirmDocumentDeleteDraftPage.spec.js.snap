// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm Document Delete Draft page should take a snapshot of the whole page 1`] = `
<div>
  <div
    class="css-1l99ixu-OuterContainer e4u1u2g0"
  >
    <div
      class="css-n57fe4-InnerContainer e4u1u2g1"
    >
      <div
        class="css-1lzr9ic-StyledRow e4u1u2g0"
      >
        <a
          class="e1vcssft0 css-zk98nb-Anchor-StyledHoc e4u1u2g0"
          href="/create-catch-certificate/catch-certificates"
        >
          Back
        </a>
      </div>
      <form
        action="/orchestration/api/v1/confirm-document-delete?c=/create-catch-certificate/catch-certificates"
        method="POST"
      >
        <div
          class="css-1lzr9ic-StyledRow e4u1u2g0"
        >
          <div
            class="css-n58wva-StyledColumn e4u1u2g0"
          >
            <div
              class="css-1lzr9ic-StyledRow e4u1u2g0"
            >
              <h1
                class="e1vcssft0 css-11nzzut-StyledHeader-StyledHoc e4u1u2g0"
              >
                <legend>
                  Are you sure you want to delete this catch certificate?
                </legend>
              </h1>
            </div>
            <div
              class="e1vcssft0 css-1phgi8s-StyledFieldset-StyledHoc e4u1u2g0"
              id="documentDelete"
            >
              <span
                class="e1vcssft0 css-3e01s4-StyledLabelText-StyledHoc e4u1u2g0"
              />
              
              <fieldset>
                <label
                  class="css-y8ql23-StyledHoc"
                  for="documentDeleteYes"
                  id="label-documentDeleteYes"
                >
                  <input
                    class="css-1xt3i3l"
                    id="documentDeleteYes"
                    name="documentDelete"
                    type="radio"
                    value="Yes"
                  />
                  <span
                    class="css-1uwxoxa"
                  >
                    Yes
                  </span>
                </label>
                <label
                  class="css-y8ql23-StyledHoc"
                  for="documentDeleteNo"
                  id="label-documentDeleteNo"
                >
                  <input
                    class="css-1xt3i3l"
                    id="documentDeleteNo"
                    name="documentDelete"
                    type="radio"
                    value="No"
                  />
                  <span
                    class="css-1uwxoxa"
                  >
                    No
                  </span>
                </label>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="css-1lzr9ic-StyledRow e4u1u2g0"
        >
          <div
            class="css-n58wva-StyledColumn e4u1u2g0"
          >
            <button
              class="css-1gc9650-StyledButton-StyledHoc e4u1u2g0"
              id="continue"
              type="submit"
            >
              Save and continue
            </button>
          </div>
        </div>
        <input
          name="journey"
          type="hidden"
          value="catchCertificate"
        />
        <input
          name="currentUri"
          type="hidden"
          value="/create-catch-certificate/catch-certificates"
        />
        <input
          name="nextUri"
          type="hidden"
          value="/create-catch-certificate/what-are-you-exporting"
        />
        <input
          name="previousUri"
          type="hidden"
          value="/create-catch-certificate/catch-certificates"
        />
      </form>
      <div
        class="css-1lzr9ic-StyledRow e4u1u2g0"
      >
        <div
          class="css-1w80l5h-StyledColumn e4u1u2g0"
        >
          <hr
            class="help-link"
          />
          <h2
            class="e1vcssft0 css-162lx7h-StyledHeader-StyledHoc e4u1u2g0"
          >
            Need help?
          </h2>
          <a
            href="undefined"
            rel="noopener noreferrer"
            target="_blank"
          >
            Get help exporting fish from the UK (gov.uk)
             
            <span
              class="govuk-visually-hidden"
            >
              (opens in new tab)
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
